File: core/src/main/java/com/yopox/screens/AbstractScreen.java
package com.yopox.screens;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.InputProcessor;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.utils.ScreenUtils;
import com.badlogic.gdx.utils.viewport.FitViewport;
import com.yopox.DeviNUPES;
import com.yopox.Util;
import com.yopox.ui.Colors;

public abstract class AbstractScreen implements InputProcessor, Screen {
    final DeviNUPES myGame;
    final SpriteBatch myBatch = new SpriteBatch();
    final OrthographicCamera myCamera = new OrthographicCamera();
    final FitViewport myViewport = new FitViewport(Util.WIDTH, Util.HEIGHT, myCamera);

    public AbstractScreen(DeviNUPES game) {
        myGame = game;
    }

    @Override
    public void show() {
        Gdx.input.setInputProcessor(this);
    }

    @Override
    public void render(float delta) {
        myBatch.setProjectionMatrix(myCamera.combined);
        ScreenUtils.clear(Colors.DEFAULT_BG);
    }

    @Override
    public void resize(int width, int height) {
        myViewport.update(width, height, true);
        myCamera.update();
    }

    @Override
    public void dispose() {
        myBatch.dispose();
    }

    @Override
    public void pause() {}

    @Override
    public void resume() {}

    @Override
    public void hide() {}

    @Override
    public boolean keyDown(int keycode) {
        return false;
    }

    @Override
    public boolean keyTyped(char character) {
        return false;
    }

    @Override
    public boolean touchDown(int screenX, int screenY, int pointer, int button) {
        return false;
    }

    @Override
    public boolean touchUp(int screenX, int screenY, int pointer, int button) {
        return false;
    }

    @Override
    public boolean touchDragged(int screenX, int screenY, int pointer) {
        return false;
    }

    @Override
    public boolean mouseMoved(int screenX, int screenY) {
        return false;
    }

    @Override
    public boolean scrolled(float amountX, float amountY) {
        return false;
    }
}

File: core/src/main/kotlin/com/yopox/AbstractScreen.kt
package com.yopox

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.InputProcessor
import com.badlogic.gdx.graphics.OrthographicCamera
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.utils.viewport.FitViewport
import com.yopox.ui.Colors
import ktx.app.KtxScreen
import ktx.app.clearScreen
import ktx.assets.disposeSafely

abstract class AbstractScreen : KtxScreen, InputProcessor {
    internal val batch = SpriteBatch()
    internal val camera = OrthographicCamera()
    internal val viewport = FitViewport(WIDTH, HEIGHT, camera)

    override fun show() {
        Gdx.input.inputProcessor = this
    }

    override fun render(delta: Float) {
        batch.projectionMatrix = camera.combined
        clearScreen(Colors.BLUE.r, Colors.BLUE.g, Colors.BLUE.b)
    }

    override fun resize(width: Int, height: Int) {
        viewport.update(width, height, true)
        camera.update()
    }

    override fun dispose() {
        batch.disposeSafely()
    }

    override fun keyDown(keycode: Int): Boolean = false

    override fun keyUp(keycode: Int): Boolean = false

    override fun touchDown(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean = false

    override fun touchUp(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean = false

    override fun touchDragged(screenX: Int, screenY: Int, pointer: Int): Boolean = false

    override fun mouseMoved(screenX: Int, screenY: Int): Boolean = false

    override fun scrolled(amountX: Float, amountY: Float): Boolean = false
}